In [2]:
import numpy as np
import pandas as pd
from bs4 import BeautifulSoup
import requests

#### 1. Identify the industry in which Canoo operates, along with its size, growth rate, trends, and key players.

In [2]:
q1=requests.get("https://www.globaldata.com/company-profile/canoo-inc/")
print(q1.status_code)

200


 meaning the website can be accessed

In [3]:
soup=BeautifulSoup(q1.text,'html.parser')
print(type(soup))

Intro=soup.p.text #navigatable string
print(Intro)
f= open("Canoo industry.txt","w")
f.write(Intro)
f.close()

<class 'bs4.BeautifulSoup'>
Canoo Inc (Canoo) is a high tech advanced mobility technology company. The company’s product offerings include true steer-by-wire, leaf spring suspension systems, advanced drivetrain systems, battery and battery management systems, electrical systems architecture, lifestyle vehicles and delivery, multi-purpose delivery vehicles and pickup. It provides electric vehicle concept, design and engineering services for auto original equipment manufacturers (OEMs). Canoo develops the Canoo Digital Ecosystem software to provide key tools for passenger vehicles and fleets with products spanning lifestyle, fleet management, security and safety, household and asset management use cases. The company serves businesses and consumers. It operates offices in California, Arkansas, Oklahama, Texas and Michigan. Canoo is headquartered in Torrance, California, the US.


In [4]:
#Canoo's Market Cap 
q2= requests.get("https://companiesmarketcap.com/canoo/marketcap/")
soup2=BeautifulSoup(q2.text,'html.parser')
marketcap= soup2.find("p",class_="mt-2")
print(marketcap.text)

As of February 2024 Canoo has a market cap of $0.15 Billion.
This makes Canoo the world's 6927th most valuable company by market cap according to our data. The market capitalization, commonly called market cap, is the total market value of a publicly traded company's outstanding shares and is commonly used to measure how much a company is worth.


In [5]:
#appending it to the txt file
f=open("Canoo industry.txt","a")
f.write(marketcap.text)


347

In [37]:
q5=requests.get("https://www.fool.com/investing/2024/01/30/where-will-canoo-stock-be-in-1-year/")
print(q5.status_code)

soup6=BeautifulSoup(q5.text,'html.parser')
#print(soup6.text)

200


In [38]:
tag=soup6.find('div',class_="article-body")
contents=tag.find_all('p')
f=open("Canoo industry.txt",'a')
for i in range(len(contents)):
    f.write(contents[i].text)
    
f.close()

In [39]:
f=open("Canoo industry.txt",'r')
f.read()

'Canoo Inc (Canoo) is a high tech advanced mobility technology company. The company’s product offerings include true steer-by-wire, leaf spring suspension systems, advanced drivetrain systems, battery and battery management systems, electrical systems architecture, lifestyle vehicles and delivery, multi-purpose delivery vehicles and pickup. It provides electric vehicle concept, design and engineering services for auto original equipment manufacturers (OEMs). Canoo develops the Canoo Digital Ecosystem software to provide key tools for passenger vehicles and fleets with products spanning lifestyle, fleet management, security and safety, household and asset management use cases. The company serves businesses and consumers. It operates offices in California, Arkansas, Oklahama, Texas and Michigan. Canoo is headquartered in Torrance, California, the US.As of February 2024 Canoo has a market cap of $0.15 Billion.\nThis makes Canoo the world\'s 6927th most valuable company by market cap accor

In [6]:
#lets check txt file
f=open("Canoo industry.txt",'r')
f.read()
f.close()

In [7]:
table = soup2.find("table", class_="table")
f = pd.DataFrame(
    columns=["Year", "Market_cap", "Change"]
)
mylist = []
for table_row in table.tbody.find_all("tr"):
    table_columns = table_row.find_all("td")

    if table_columns != []:
        year = table_columns[0].text.strip()
        market_cap = table_columns[1].text.strip()
        change = table_columns[2].text.strip()

        mylist.append([year,market_cap,change])

df = pd.DataFrame(
    mylist,
    columns=["Year", "Market_cap", "Change"]
)



In [8]:
df

Year Market_cap   Change
0  2024    $0.15 B  -22.38%
1  2023    $0.20 B  -52.36%
2  2022    $0.42 B  -76.98%
3  2021    $1.84 B  -43.32%
4  2020    $3.25 B

In [9]:
#saving as csv file
df.to_csv("Canoo_market_cap.csv")

In [10]:
#Canoo Earnings
q3=requests.get("https://companiesmarketcap.com/canoo/earnings/")
soup3=BeautifulSoup(q3.text,'html.parser')


In [11]:
table = soup3.find("table", class_="table")
f = pd.DataFrame(
    columns=["Year", "Earnings", "Diff"]
)
mylist = []
for table_row in table.tbody.find_all("tr"):
    table_columns = table_row.find_all("td")

    if table_columns != []:
        year = table_columns[0].text.strip()
        earnings = table_columns[1].text.strip()
        diff = table_columns[2].text.strip()

        mylist.append([year,earnings,diff])

df2 = pd.DataFrame(
    mylist,
    columns=["Year", "Earnings", "diff"]
)


In [12]:
df2

Year   Earnings     diff
0  2023 (TTM)   -$0.36 B  -28.66%
1        2022    -$0.5 B   10.59%
2        2021   -$0.45 B  467.03%
3        2020  -$79.35 M

In [13]:
df

Year Market_cap   Change
0  2024    $0.15 B  -22.38%
1  2023    $0.20 B  -52.36%
2  2022    $0.42 B  -76.98%
3  2021    $1.84 B  -43.32%
4  2020    $3.25 B

In [15]:
df_new=pd.merge(df,df2,on="Year",how="left")
df_new

Year Market_cap   Change   Earnings     diff
0  2024    $0.15 B  -22.38%        NaN      NaN
1  2023    $0.20 B  -52.36%        NaN      NaN
2  2022    $0.42 B  -76.98%    -$0.5 B   10.59%
3  2021    $1.84 B  -43.32%   -$0.45 B  467.03%
4  2020    $3.25 B           -$79.35 M

In [16]:
#saving it to the csv file
df_new.to_csv("Canoo_market_earnings.csv")

### Canoo's Competitors


In [17]:
q3=requests.get("https://www.macrotrends.net/stocks/charts/GOEV/canoo/market-cap")
print(q3.status_code)

soup4=BeautifulSoup(q3.text,'html.parser')
print(type(soup4)) 

403
<class 'bs4.BeautifulSoup'>


 The above website cannot be acessed as it has 403 as status code

In [3]:
q3=requests.get("https://incfact.com/company/canoo-torrance-ca/#revenue")
print(q3.status_code)

soup5= BeautifulSoup(q3.text,'lxml')
print(type(soup5)) 

200
<class 'bs4.BeautifulSoup'>


In [28]:
table= soup5.find_all('table',class_="table outputtable")[1]
print(table.text)
table_rows=table.find_all("tr")
#print(table_rows)

mylist=[]
f=pd.DataFrame(
    columns=['Company','Headquaters']
)
for row in table_rows:
    table_columns = row.find_all("td")

    if table_columns != []:
        company = table_columns[0].text.strip()
        Headquaters = table_columns[1].text.strip()

        mylist.append([company,Headquaters])

df3 = pd.DataFrame(
    mylist,
    columns=['Company','Headquaters']
)



Company
HeadquartersRevenue ($ MM)TESLAPalo Alto, CA100 GENERAL MOTORS HOLDINGSDetroit, MI99 UAW-FORD VOLUNTARY EMPLOYEES BEDetroit, MI63 FCA USAuburn Hills, MI59 AMERICAN HONDA MOTORMarysville, OH45 NORTHROP GRUMMAN SYSTEMSFairfax, VA43 EATONCleveland, OH39 BOEINGSeattle, WA35 


In [29]:
df3

Company       Headquaters
0                            TESLA     Palo Alto, CA
1          GENERAL MOTORS HOLDINGS       Detroit, MI
2  UAW-FORD VOLUNTARY EMPLOYEES BE       Detroit, MI
3                           FCA US  Auburn Hills, MI
4             AMERICAN HONDA MOTOR    Marysville, OH
5         NORTHROP GRUMMAN SYSTEMS       Fairfax, VA
6                            EATON     Cleveland, OH
7                           BOEING       Seattle, WA

In [31]:
table= soup5.find_all('table',class_="table outputtable")[2]
print(table.text)
table_rows=table.find_all("tr")
for row in table_rows:
    table_columns = row.find_all("td")

    if table_columns != []:
        company = table_columns[0].text.strip()
        Headquaters = table_columns[1].text.strip()

        mylist.append([company,Headquaters])

df3 = pd.DataFrame(
    mylist,
    columns=['Company','Headquaters']
)



Company
HeadquartersRevenue ($ MM)CANOO TECHNOLOGIESTorrance, CA100 LISI HOLDING NORTH AMERICATorrance, CA65 ROBINSON HELICOPTERTorrance, CA54 DAICEL AMERICA HOLDINGSTorrance, CA44 MOTORCAR PARTS OF AMERICATorrance, CA23 GARRETT TRANSPORTATION ITorrance, CA11 


In [35]:
df3

Company       Headquaters
0                             TESLA     Palo Alto, CA
1           GENERAL MOTORS HOLDINGS       Detroit, MI
2   UAW-FORD VOLUNTARY EMPLOYEES BE       Detroit, MI
3                            FCA US  Auburn Hills, MI
4              AMERICAN HONDA MOTOR    Marysville, OH
5          NORTHROP GRUMMAN SYSTEMS       Fairfax, VA
6                             EATON     Cleveland, OH
7                            BOEING       Seattle, WA
8                CANOO TECHNOLOGIES      Torrance, CA
9        LISI HOLDING NORTH AMERICA      Torrance, CA
10              ROBINSON HELICOPTER      Torrance, CA
11          DAICEL AMERICA HOLDINGS      Torrance, CA
12        MOTORCAR PARTS OF AMERICA      Torrance, CA
13         GARRETT TRANSPORTATION I      Torrance, CA

In [36]:
df3.to_csv("canoo competitors.csv")